### PR TITLE
test: Required encryption context CMM integration tests

### PR DESCRIPTION
**Pull request by lucasmcdonald3**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: test/mpl/integ/test_required_ec_cmm.py [hunk 1]
```diff
@@ -0,0 +1,737 @@
...
+    encryption_context
+):
+    # This test needs >1 item to supply as required encryption context
+    if len(encryption_context) < 1:
+        return
+
+    # Grab one item from encryption_context to supply as reproduced EC
+    one_k, one_v = next(iter(encryption_context.items()))
+    reproduced_ec = {one_k: one_v}
+    # Given: one required encryption context (REC) key
```
Comment (seebees) [inline test/mpl/integ/test_required_ec_cmm.py:244]: I love these comments. Is it possible for you to use `duvet` to align these with https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/test-vectors/complete-vectors/required-encryption-context-cmm.md?

Comment (lucasmcdonald3) [reply test/mpl/integ/test_required_ec_cmm.py:244]: Chatted and decided this isn't worth it.

File: test/mpl/integ/test_required_ec_cmm.py [hunk 2]
```diff
@@ -0,0 +1,737 @@
...
+# HAPPY CASE 2
+# On Encrypt we will only write one encryption context key value to the header
+# we will then supply only what we didn't write wth no required ec cmm,
+# This test case is checking that the default cmm is doing the correct filtering
+def test_GIVEN_RECCMM_with_one_REC_key_on_encrypt_AND_default_CMM_with_valid_reproduced_EC_on_decrypt_WHEN_supply_reproduced_EC_with_REC_key_on_decrypt_THEN_decrypt_matches_plaintext( # noqa pylint: disable=line-too-long
+    encryption_context
+):
+    # This test needs >1 item to supply as required encryption context
+    if len(encryption_context) < 1:
+        return
```
[the remark this answers is not in the document]
Comment (lucasmcdonald3) [reply test/mpl/integ/test_required_ec_cmm.py]: Refactored this, the conditions are now implicit in the test parameters.

**Review by seebees — APPROVED** [review]
Looks good!

**Pull request merged**